build/configs/toolchain_external.cmake

```diff
@@ -17,4 +17,4 @@ include(CMakeForceCompiler)
 set(CMAKE_SYSTEM_NAME EXTERNAL)
 set(CMAKE_SYSTEM_PROCESSOR "${EXTERNAL_CMAKE_SYSTEM_PROCESSOR}")
 
-CMAKE_FORCE_C_COMPILER(${EXTERNAL_CMAKE_C_COMPILER} GNU)
+CMAKE_FORCE_C_COMPILER(${EXTERNAL_CMAKE_C_COMPILER} ${EXTERNAL_CMAKE_C_COMPILER_FAMILY})
```

Comment:
Actually, the second parameter of `CMAKE_FORCE_C_COMPILER` is called <compiler-id> and sets the `CMAKE_C_COMPILER_ID` cmake internal variable. Thus, it is preferrable to call the external variant `EXTERNAL_CMAKE_C_COMPILER_ID` (instead of `_FAMILY`).

Moreover, the introduction of this new variable breaks all the current ports that use this external cmake file, since the Makefiles under the targets/ directory don't specify value for `EXTERNAL_CMAKE_C_COMPILER_ID`. Please, update those files.